pkg/config/template.go

```diff
@@ -85,6 +88,22 @@ func NewTemplateCMManager(cmclient corev1.ConfigMapInterface, prNumber int, logg
 	}
 }
 
+func (c *TemplateCMManager) createCM(cm *v1.ConfigMap) error {
+	l := &cm.ObjectMeta.Labels
```

Comment:
I think you don't need `&` here. Also, you will be able to update the value below without passing the pointer.  However, you can also `make()` it.

Reply:
Would the assignment inside the `if` also work if it weren't a pointer?

Reply:
it should

Reply:
Doesn't look like it (which follows my understanding of how this should work, but there might be some `map` magic I am not aware of):

```go
package main

import "fmt"

func main() {
    type S struct {m map[int]int}
    s := S{}
    n := s.m
    fmt.Println(n == nil)
    if n == nil {
        n = map[int]int{}
    }
    n[0] = 1
    fmt.Printf("%#v %#v\n", s.m, n)
    t := S{}
    p := &t.m
    fmt.Println(*p == nil)
    if *p == nil {
        *p = map[int]int{}
    }
    (*p)[0] = 1
    fmt.Printf("%#v %#v\n", t.m, p)
}
```

```
true
map[int]int(nil) map[int]int{0:1}
true
map[int]int{0:1} &map[int]int{0:1}
```